crypto/md2/md2_dgst.c

```diff
@@ -70,6 +77,10 @@ const char *MD2_options(void)
 
 int MD2_Init(MD2_CTX *c)
 {
+    if (FIPS_mode()) {
+        FIPSerr(ERR_LIB_FIPS, FIPS_R_NON_FIPS_METHOD);
+        return 0;
```

Comment:
In `MD*_Init()` and the like, can we perhaps allow the context initialization to complete, and then return error? Would be very slightly safer in terms of not leaving uninitialized data that an application may use somehow. I hope it's not a FIPS violation merely to initialize a struct with constants or zeroize it?

Reply:
I don't think that's the right approach. Functions that return failure should *never* modify incoming data. In Samba, this is a golden rule. We only copy initialized data into "out" parameters just before returning success.

On failure return, no change should be made to "out" parameters. I think you'll find this is the rule in glibc and most other libraries too.

Reply:
Resolved with the follow-up compromise patch.